lib/EnzymeCore/src/EnzymeCore.jl

```diff
@@ -239,6 +239,8 @@ end
 
 function tape_type end
 
+function parent_job_for_tape_type end
```

Comment:
I don't quite follow the semantics here, can you write a docstring discussing intended usage?

Reply:
Its pretty much https://github.com/JuliaGPU/CUDA.jl/blob/8e25fc1ada7f63e2e9b93785b451a751e58a20e1/ext/EnzymeCoreExt.jl#L9 

for the extended version of tape_type introduced in #1104 we need to be able to construct a parent_job explicitly. There is a question who should own that API. Originally I had it in KA, but it makes more sense in EnzymeCore, since otherwise we can never call the new `tape_type` function.